eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/BackPropagatedRSWriter.java

```diff
@@ -0,0 +1,259 @@
+                                                             ClusteredVariantEntity.class);
+        BulkOperations dbsnpCVEInserts = mongoTemplate.bulkOps(BulkOperations.BulkMode.UNORDERED,
+                                                               DbsnpClusteredVariantEntity.class);
+
+        // Updates to Submitted Variant Entity (SVE) collection with the RS created above
+        BulkOperations evaSVEUpdates = mongoTemplate.bulkOps(BulkOperations.BulkMode.UNORDERED,
+                                                             SubmittedVariantEntity.class);
+        BulkOperations dbsnpSVEUpdates = mongoTemplate.bulkOps(BulkOperations.BulkMode.UNORDERED,
+                                                               DbsnpSubmittedVariantEntity.class);
+
+        // Inserts to Submitted Variant Operation Entity (SVOE) collection recording the RS updates made above
+        BulkOperations evaSVOEInserts = mongoTemplate.bulkOps(BulkOperations.BulkMode.UNORDERED,
+                                                              SubmittedVariantOperationEntity.class);
+        BulkOperations dbsnpSVOEInserts = mongoTemplate.bulkOps(BulkOperations.BulkMode.UNORDERED,
+                                                                DbsnpSubmittedVariantOperationEntity.class);
+
+
+        int numEvaRSAssignments = 0;
+        int numDbsnpRSAssignments = 0;
+        int numEVARSInserts = 0;
+        int numDbsnpRSInserts = 0;
+
+        for (SubmittedVariantEntity submittedVariantEntity: submittedVariantEntitiesInOriginalAssemblyWithNoRS) {
+            Long ssIDToAssignRS = submittedVariantEntity.getAccession();
+            if (ssInRemappedAssemblyGroupedByID.containsKey(ssIDToAssignRS)) {
+                BulkOperations bulkSVEUpdates, bulkSVOEInserts, bulkCVEInserts;
+                if (clusteringWriter.isEvaSubmittedVariant(submittedVariantEntity)) {
+                    bulkSVEUpdates = evaSVEUpdates;
+                    bulkSVOEInserts = evaSVOEInserts;
+                    numEvaRSAssignments += 1;
+                } else {
+                    bulkSVEUpdates = dbsnpSVEUpdates;
+                    bulkSVOEInserts = dbsnpSVOEInserts;
+                    numDbsnpRSAssignments += 1;
+                }
+                ClusteredVariantEntity newRSRecordForOriginalAssembly =
+                        getSuitableRSFromRemappedSS(ssInRemappedAssemblyGroupedByID, submittedVariantEntity);
+                Long rsToAssign = newRSRecordForOriginalAssembly.getAccession();
+
+                if (clusteringWriter.getClusteredVariantCollection(rsToAssign).equals(ClusteredVariantEntity.class)) {
+                    bulkCVEInserts = evaCVEInserts;
+                    numEVARSInserts += 1;
+                } else {
+                    bulkCVEInserts = dbsnpCVEInserts;
+                    numDbsnpRSInserts += 1;
+                }
+                bulkCVEInserts.insert(newRSRecordForOriginalAssembly);
```

Comment:
I though we were not supposed to insert back the RS but simply annotate the SS.

Reply:
yes, I also thought we would be back-filing unclustered SS ids from old assemblies using RS ids created in newer assemblies but not creating RS ids for the old assemblies

Reply:
Having separate RS records in each assembly with their own hashes facilitates clean separation of records across assemblies and could lessen maintenance headaches when we archive accessioning data in older assemblies.